storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java

```diff
@@ -4654,6 +4714,55 @@ public ClusterSummary getClusterInfo() throws AuthorizationException, TException
         }
     }
 
+    @Override
+    public List<TopologySummary> getTopologySummaries() throws AuthorizationException, TException {
+        try {
+            getTopologySummariesCalls.mark();
+            checkAuthorization(null, null, "getTopologySummaries");
+            return getTopologySummariesImpl();
+        } catch (Exception e) {
+            LOG.warn("Get TopologySummary info exception.", e);
+            if (e instanceof TException) {
+                throw (TException) e;
+            }
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public TopologySummary getTopologySummaryByName(String name) throws AuthorizationException, TException {
+        try {
+            getTopologySummaryByNameCalls.mark();
+            checkAuthorization(null, null, "getTopologySummaries");
+            IStormClusterState state = stormClusterState;
+            String topoId = state.getTopoId(name).orElseThrow(() -> new WrappedNotAliveException(name + " is not alive"));
+            return getTopologySummary(topoId, state.topologyBases().get(topoId));
+        } catch (Exception e) {
+            LOG.warn("Get TopologySummaryByName info exception.", e);
+            if (e instanceof TException) {
+                throw (TException) e;
+            }
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public TopologySummary getTopologySummaryById(String id) throws AuthorizationException, TException {
+        try {
+            getTopologySummaryByIdCalls.mark();
+            IStormClusterState state = stormClusterState;
+            StormBase base = state.topologyBases().get(id);
```

Comment:
`base` can be null if this topology doesn't exist. Then there will be NullPointerException in `getTopologySummary(id, base)` . Is this intended?

Reply:
> `base` can be null if this topology doesn't exist. Then there will be NullPointerException in `getTopologySummary(id, base)` . Is this intended?

We need to check `base==null` here